yarn-project/prover-client/src/bb/execute.ts

```diff
@@ -0,0 +1,294 @@
+    .catch(_ => false);
+  if (!binaryPresent) {
+    return { status: BB_RESULT.FAILURE, reason: `Failed to find bb binary at ${pathToBB}` };
+  }
+
+  // We are now going to generate the key
+  try {
+    // Clear up the circuit output directory removing anything that is there
+    await fs.rm(circuitOutputDirectory, { recursive: true, force: true });
+    await fs.mkdir(circuitOutputDirectory, { recursive: true });
+
+    // Write the bytecode to the working directory
+    await fs.writeFile(bytecodePath, bytecode);
+
+    // args are the output path and the input bytecode path
+    const args = ['-o', outputPath, '-b', bytecodePath];
+    const timer = new Timer();
+    const result = await executeBB(pathToBB, `write_${key}`, args, log);
+    const duration = timer.ms();
+    // Cleanup the bytecode file
+    await fs.rm(bytecodePath, { force: true });
+    if (result == BB_RESULT.SUCCESS) {
+      // Store the bytecode hash so we don't need to regenerate at a later time
+      await fs.writeFile(bytecodeHashPath, bytecodeHash);
+      return { status: BB_RESULT.SUCCESS, duration, path: outputPath };
+    }
+    // Not a great error message here but it is difficult to decipher what comes from bb
+    return { status: BB_RESULT.FAILURE, reason: `Failed to generate key` };
+  } catch (error) {
+    return { status: BB_RESULT.FAILURE, reason: `${error}` };
+  }
+}
+
+/**
+ * Used for generating either a proving or verification key, will exit early if the key already exists
+ * It assumes the provided working directory is one where the caller wishes to maintain a permanent set of keys
+ * It is not considered a temporary directory
+ * @param pathToBB - The full path to the bb binary
+ * @param workingDirectory - The directory into which the key should be created
+ * @param circuitName - An identifier for the circuit
+ * @param compiledCircuit - The compiled circuit
+ * @param key - The type of key, either 'pk' or 'vk'
+ * @param log - A logging function
+ * @returns The path to the key, or undefined.
+ */
+// export async function generateKey(
+//   pathToBB: string,
+//   workingDirectory: string,
+//   circuitName: string,
+//   compiledCircuit: NoirCompiledCircuit,
+//   key: 'pk' | 'vk',
+//   log: LogFn,
+// ) {
+//   const { result, duration, outputPath } = await generateKeyForNoirCircuit(
+//     pathToBB,
+//     workingDirectory,
+//     circuitName,
+//     compiledCircuit,
+//     key,
+//     log,
+//   );
+//   if (result.status === BB_RESULT.FAILURE) {
+//     log(`Failed to generate ${key} key for circuit ${circuitName}, reason: ${result.reason}`);
+//     return;
+//   }
+//   if (result.status === BB_RESULT.ALREADY_PRESENT) {
+//     log(`Key ${key} for circuit ${circuitName} was already present`);
+//     return outputPath;
+//   }
+//   const stats = await fs.stat(outputPath);
+//   log(`Key ${key} for circuit ${circuitName} generated in ${duration} ms, size: ${stats.size / (1024 * 1024)} MB`);
+//   return outputPath;
+// }
```

Comment:
Is this still needed or could it be deleted?

Reply:
Removed